pxr/imaging/hdx/shaders/renderPass.glslfx

```diff
@@ -192,11 +261,70 @@ void RenderOutput(vec4 Peye, vec3 Neye, vec4 color, vec4 patchCoord)
     int instanceId = GetDrawingCoord().instanceIndex[0];
     instanceIdOut = IntToVec4(instanceId);
 
-    elementIdOut = IntToVec4(GetElementID());
-    edgeIdOut = IntToVec4(GetPrimitiveEdgeId());
-    pointIdOut = IntToVec4(GetPointId());
+    int elementId = GetElementID();
+    int edgeId = GetPrimitiveEdgeId();
+    int pointId = GetPointId();
+
+    elementIdOut = IntToVec4(elementId);
+    edgeIdOut = IntToVec4(edgeId);
+    pointIdOut = IntToVec4(pointId);
     
     neyeOut = IntToVec4(hd_vec4_2_10_10_10_set(vec4(Neye,0)));
+
+    // add the item to the pick buffer, but only unique
+#if defined(HD_HAS_PickBuffer)
+
+    // if the pick buffer is not initialized, we are done 
```

Comment:
Somewhere (here, pickTask.cpp, elsewhere?) it would be fantastic to have both a block comment about the buffer layout, and a block comment about the algorithm you're using to write the entry out, since they're both tricky to pick up from reading the code.